packages/hoppscotch-js-sandbox/src/apis/expect/utils.ts

```diff
@@ -0,0 +1,338 @@
+            message: `Expected the array to${
+              negated ? " not" : ""
+            } be of length '${expectedLength}'`,
+          })
+        }
+      } else {
+        currTestStack[currTestStack.length - 1].expectResults.push({
+          status: "error",
+          message: `Argument for toHaveLength should be a number`,
+        })
+      }
+
+      return { value: vm.undefined }
+    }
+  )
+
+  const toIncludeHandle = vm.newFunction("toInclude", (needleHandle) => {
+    const expectedVal = vm.dump(needleHandle)
+
+    if (!(Array.isArray(expectVal) || typeof expectVal === "string")) {
+      currTestStack[currTestStack.length - 1].expectResults.push({
+        status: "error",
+        message: `Expected toInclude to be called for an array or string`,
+      })
+
+      return { value: vm.undefined }
+    }
+
+    if (expectedVal === null) {
+      currTestStack[currTestStack.length - 1].expectResults.push({
+        status: "error",
+        message: `Argument for toInclude should not be null`,
+      })
+
+      return { value: vm.undefined }
+    }
+
+    if (expectedVal === undefined) {
+      currTestStack[currTestStack.length - 1].expectResults.push({
+        status: "error",
+        message: `Argument for toInclude should not be undefined`,
+      })
+
+      return { value: vm.undefined }
+    }
+
+    let assertion = expectVal.includes(expectedVal)
+    if (negated) assertion = !assertion
+
+    const expectValPretty = JSON.stringify(expectVal)
+    const expectedValPretty = JSON.stringify(expectedVal)
+
+    if (assertion) {
+      currTestStack[currTestStack.length - 1].expectResults.push({
+        status: "pass",
+        message: `Expected ${expectValPretty} to${
+          negated ? " not" : ""
+        } include ${expectedValPretty}`,
+      })
+    } else {
+      currTestStack[currTestStack.length - 1].expectResults.push({
+        status: "fail",
+        message: `Expected ${expectValPretty} to${
+          negated ? " not" : ""
+        } include ${expectedValPretty}`,
+      })
+    }
+
+    return { value: vm.undefined }
+  })
+
+  vm.setProp(resultHandle, "toBe", toBeFnHandle)
+  vm.setProp(resultHandle, "toBeLevel2xx", toBeLevel2xxHandle)
+  vm.setProp(resultHandle, "toBeLevel3xx", toBeLevel3xxHandle)
+  vm.setProp(resultHandle, "toBeLevel4xx", toBeLevel4xxHandle)
+  vm.setProp(resultHandle, "toBeLevel5xx", toBeLevel5xxHandle)
+  vm.setProp(resultHandle, "toBeType", toBeTypeHandle)
+  vm.setProp(resultHandle, "toHaveLength", toHaveLengthHandle)
+  vm.setProp(resultHandle, "toInclude", toIncludeHandle)
+
+  vm.defineProp(resultHandle, "not", {
+    get: () => {
+      return createExpectation(vm, expectVal, !negated, currTestStack)
+    },
+  })
+
+  toBeFnHandle.dispose()
+  toBeLevel2xxHandle.dispose()
+  toBeLevel3xxHandle.dispose()
+  toBeLevel4xxHandle.dispose()
+  toBeLevel5xxHandle.dispose()
+  toBeTypeHandle.dispose()
+  toHaveLengthHandle.dispose()
+  toIncludeHandle.dispose()
```

Comment:
Can't we use those new utility functions to simplify this ?

Reply:
Yes, we can use, I missed this.